File: src/tests/Console.Tests/RegionTableMetadata.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace ExtenderRegion
{


		public class RegionTableMetadata : TableMetadata
		{
			private DatabaseField[] _fields;

			public RegionTableMetadata()
			{
					_fields = new DatabaseField[2];
 _fields[0] = new DatabaseField(DbType.Int32,"RegionID",true,true,null);
 _fields[1] = new DatabaseField(DbType.String,"RegionDescription",false,false,null);
 
this.tableName = "Region";


}


			public override DatabaseField[] TableFields 
			{
				get{ return _fields;}
				set{_fields = value;}
			}

public System.Int32 RegionID
{
    get
    {
          return (System.Int32) (this.GetField("RegionID")).fieldValue;
    }

    set
    {
          this.SetFieldValue("RegionID", value);
    }
}


public System.String RegionDescription
{
    get
    {
          return (System.String) (this.GetField("RegionDescription")).fieldValue;
    }

    set
    {
          this.SetFieldValue("RegionDescription", value);
    }
}

}
}

File: src/tests/DataBlock Sample Application/VB.NET/Regions/RegionTableMetadata.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock
Namespace ExtenderRegion

    Public Class RegionTableMetadata
        Inherits TableMetadata
        Private _fields As DatabaseField()

        Public Sub New()
            _fields = New DatabaseField(1) {}
            _fields(0) = New DatabaseField(DbType.Int32, "RegionID", True, False, Nothing)
            _fields(1) = New DatabaseField(DbType.String, "RegionDescription", False, False, Nothing)
            Me.tableName = "Region"
        End Sub

        Public Overloads Overrides Property TableFields() As DatabaseField()
            Get
                Return _fields
            End Get
            Set(ByVal Value As DatabaseField())
                _fields = value
            End Set
        End Property

        Public Property RegionID() As System.Int32
            Get
                Return CType((Me.GetField("RegionID")).fieldValue, System.Int32)
            End Get
            Set(ByVal Value As System.Int32)
                Me.SetFieldValue("RegionID", value)
            End Set
        End Property

        Public Property RegionDescription() As System.String
            Get
                Return CType((Me.GetField("RegionDescription")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("RegionDescription", value)
            End Set
        End Property
    End Class
End Namespace
